pkg/planner/core/indexmerge_path.go

```diff
@@ -391,19 +479,48 @@ func (ds *DataSource) generateIndexMergeAndPaths(normalPathCnt int) *util.Access
 		if ds.possibleAccessPaths[i].IsTablePath() {
 			continue
 		}
+		// since this code path is only for normal index, skip mv index here.
+		if ds.possibleAccessPaths[i].Index.MVIndex {
+			continue
+		}
 		if !ds.isSpecifiedInIndexMergeHints(originalPath.Index.Name.L) {
 			continue
 		}
 		// If the path contains a full range, ignore it.
 		if ranger.HasFullRange(originalPath.Ranges, false) {
 			continue
 		}
+		if composedWithMvIndex {
+			// case:
+			// idx1: mv(c, cast(`a` as signed array))
+			// idx2: idx(c)
+			// idx3: idx(c, d)
+			// for condition: (1 member of a) AND (c = 1) AND (d = 2), we should pick idx1 and idx3,
+			// since idx2's access cond has already been covered by idx1.
+			containRelation := true
+			for _, access := range originalPath.AccessConds {
```

Comment:
Should we order all paths first? Otherwise if the input order is `idx2`, `idx1` `idx3`, the filter logic seems not working.

Reply:
But maybe it's a little complex to order them...

Reply:
you mean when c is not used, which index below should be used to build normal index partial path？yes, it a question, for now it's random, first coming first usage, map it as used. so idx3 is ignored, unless there is a new condition d=1 remained.

idx2: idx(c)
idx3: idx(c, d)